### ISSUE TITLE
[FSR] Discovery for aria-describedby for back/continue cancel/add buttons to provide more context to screen readers

### ISSUE THREAD

**Issue by josephrlee** [department-of-veterans-affairs/va.gov-team#69828]
## Description

During an AT study, we noticed a blind participant accidentally clicking `back` instead of `continue` and then getting confused about what page they were on in the VA 5655 (FSR). This got me thinking about a demo shared at FormFest last week; one of the devs from Code From America showcased the `aria-descirbedby` attribute to give more context to various elements for screenreaders. I'm thinking about utilizing it on the `back/continue` buttons to help veterans who use AT have more context on where they are going in the form and tie it to the h3 on the page.

Given: A blind user navigates the FSR
When: A blind user clicks `back/continue` 
Then: The screenreader will read the `h3` on the previous page when clicking `back` and the `h3` on the next page when clicking `continue` 

## Tasks
- [x] Start thread on Slack to see if this is a good idea
- [x] Work with FE team to determine LOE
- [x] Come up with working concept to test

## Acceptance criteria

- [ ] A screenreader user will be able to get more context when clicking `back/continue` or `cancel/add` on the FSR using the `aria-describedby` attribute 

## Testing
- [ ] N/A if non development work

Testing procedure:
If additional testing steps or credentials to perform testing are needed list them here

---

## Ticket creation

<details><summary>Ticket requirements</summary>

### Required to be "Ready"
- [x] Description written
- [x] Tasks defined
- [x] Acceptance criteria written
- [x] Testing defined
- [x] Add Labels
- [ ] Attach to an Epic
- [x] Estimate Points
</details>

**Comment by josephrlee** [comment]
@Tom-Davis @HeatherRienks, I created a new discovery ticket to improve the FSR navigation for screenreader users. I'm not sure what epic this should go in, though. Any thoughts? Thanks!

**Comment by josephrlee** [comment]
Here's the slack thread: https://dsva.slack.com/archives/C8E985R32/p1699916257494799

**Comment by josephrlee** [comment]
Didn't get around to this ticket as the bulk of the time was spent on payment history/diary code/data dictionary work. I will be bringing this in this sprint, though!

**Comment by HeatherRienks** [comment]
Per Sprint Planning, moving back to Ready while we focus on MVP - VBA Payment History.

**Comment by HeatherRienks** [comment]
This ticket seems to have been accidentally pulled into the sprint and it subsequently rolled over! It should have been moved to "ready" (see comment from 6/18/24).

**Comment by josephrlee** [comment]
After pairing with @amponce, we decided not to move forward with it. We have a working concept, but it can only be implemented for our custom button group, and making it unique to every page would be too invasive and laborious.

If our form were simpler and didn't have conditional logic and SW logic, this would be a great addition. However, because it is so complex, structuring and dynamically implementing it would require significant effort. Trying to account for the `cancel/edit or cancel/update` or the `back to review` buttons makes this much more complex. 

Our working concept would add `Back to previous page` or `Continue to next page` to the [ButtonGroup.jsx](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/financial-status-report/components/shared/ButtonGroup.jsx) component:

```
const Button = ({ label, onClick, isSubmitting, isSecondary }) => {
  let ariaDescribedBy;
  const lowerLabel = label.toLowerCase();

  if (lowerLabel.includes('back')) {
    ariaDescribedBy = 'Go back to previous page';
  } else if (lowerLabel.includes('continue')) {
    ariaDescribedBy = 'Continue to next page';
  } else {
    ariaDescribedBy = '';
  }
  if (isSubmitting) {
    return (
      <VaButton
        data-testid="custom-button-group-button"
        text={label}
        onClick={onClick}
        submit={isSubmitting}
        secondary={isSecondary}
        message-aria-describedby={ariaDescribedBy}
      />
    );
  }

  return (
    <VaButton
      data-testid="custom-button-group-button"
      text={label}
      onClick={onClick}
      secondary={isSecondary}
      message-aria-describedby={ariaDescribedBy}
    />
  );
};
```
[But based on the DS guidance](https://design.va.gov/components/button/#additional-guidance-for-va-1), our solution wouldn't provide enough additional context to be useful:
![image](https://github.com/user-attachments/assets/53916f3a-18b0-4d03-8a97-16e778e135ab)

Here is a recording showing our solution, which falls into the *When not to use* category:

https://github.com/user-attachments/assets/45385b54-bcf9-4a59-aa8a-1907b13feab4

This solution would only target the pages using the `ButtonGroup` component, leaving the remaining pages using the default form system buttons unchanged. Thus, it would only affect 26 out of the 70~ pages we have.
